crates/gpui/src/text_system/line.rs

```diff
@@ -103,13 +106,16 @@ impl WrappedLine {
         &self,
         origin: Point<Pixels>,
         line_height: Pixels,
+        align: TextAlign,
         window: &mut Window,
         cx: &mut App,
     ) -> Result<()> {
         paint_line(
             origin,
             &self.layout.unwrapped_layout,
             line_height,
+            align,
+            self.layout.wrap_width,
```

Comment:
The `whitespace_normal` can works, because of the `wrap_width` have value.

This is by TextLayout to measured known_dimensions:

https://github.com/zed-industries/zed/blob/1301c41ceaf19f4679b5f8d137747c3716762edc/crates/gpui/src/elements/text.rs#L287